pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -532,6 +532,9 @@ private PGStream enableSSL(PGStream pgStream, SslMode sslMode, Properties info,
 
     LOGGER.log(Level.FINEST, " FE=> SSLRequest");
 
+    // set the time out to 2seconds
+    int currentTimeout = pgStream.getSocket().getSoTimeout();
+    pgStream.getSocket().setSoTimeout(2000);
```

Comment:
I wonder if 2sec would give us flaky timeout errors caused by gc pauses or something else.

Should we have a property to make it configurable?

Reply:
> I wonder if 2sec would give us flaky timeout errors caused by gc pauses or something else.
One of those questions only experience will answer. Probably a little too short though
> 
> Should we have a property to make it configurable?

Probably...